backend/rest_api_service/crud.py

```diff
@@ -10,6 +11,61 @@ async def get_all_users(db: Session) -> list[Type[DBUser]] | None:
     return users
 
 
+
+ 
+async def get_user_by_id(db: Session, user_id: int) -> models.DBUser | None:
+    user = db.query(models.DBUser).filter(models.DBUser.user_id == user_id).first()
+    print(user)
+    
+    return user
+
+async def get_user_by_username(db: Session, username: str) -> models.DBUser | None:
+    user = db.query(models.DBUser).filter(models.DBUser.username == username).first()
+    print(user)
+    
+    return user
+
+async def get_all_events_by_user_id(db: Session, user_id: int) -> models.DBUser | None:
+    user = db.query(models.DBUser).options(joinedload(models.DBUser.events_created)).filter(models.DBUser.user_id == user_id).first()
+    return user
+
+
+async def get_all_friends_by_user_id(db: Session, user_id: int) -> models.DBUser | None:
+    user = db.query(models.DBUser).filter(models.DBUser.user_id == user_id).first()
+    if user:
+        return user.friends
+    return None
+
+async def create_user(db: Session, username: str, email: str, password_hash: str, birthday: date, is_active: bool = True):
+    new_user = DBUser(
+        username=username,
+        email=email,
+        password_hash=password_hash,
+        birthday=birthday,
+        created_at=date.today(),
+        last_login=date.today(),
+        update_date=date.today(),
+        is_active=is_active
+    )
+    db.add(new_user)
+    db.commit()
+    db.refresh(new_user)
+    return new_user
+
+async def delete_user(db: Session, user_id: int) -> bool:
```

Comment:
usuwanie rekordów z DBUserFriendship spoko, tylko chyba jeszcze brakuje usunięcia rekordów z DBUser

Reply:
Co do tego, to Kacper powiedział, że ogarnie, więc całkiem to usunę.

Reply:
dobra